plugin/src/main/java/org/openrewrite/gradle/isolated/DefaultProjectParser.java

```diff
@@ -1335,7 +1355,33 @@ private SourceFile logParseErrors(SourceFile source) {
         return source;
     }
 
-    static boolean isExcluded(Collection<PathMatcher> exclusions, Path path) {
+    static boolean isExcluded(@Nullable Repository repository, Collection<PathMatcher> exclusions, Path path) {
+        if (repository != null) {
+            String repoRelativePath = PathUtils.separatorsToUnix(path.toString());
+            if (repoRelativePath.isEmpty()) {
+                return false;
+            }
+
+            try (TreeWalk walk = new TreeWalk(repository)) {
+                walk.addTree(new FileTreeIterator(repository));
+                walk.setFilter(PathFilterGroup.createFromStrings(repoRelativePath));
+                while (walk.next()) {
+                    WorkingTreeIterator workingTreeIterator = walk.getTree(0, WorkingTreeIterator.class);
+                    if (walk.getPathString().equals(repoRelativePath)) {
+                        return workingTreeIterator.isEntryIgnored();
+                    }
+                    if (workingTreeIterator.getEntryFileMode().equals(FileMode.TREE)) {
+                        if (workingTreeIterator.isEntryIgnored()) {
+                            return true;
+                        } else {
+                            walk.enterSubtree();
+                        }
+                    }
+                }
+            } catch (IOException e) {
+                throw new UncheckedIOException(e);
+            }
+        }
```

Comment:
Would it make sense to do the PathMatcher checks first? I imagine those are cheaper to do, and quite likely to match if configured.